### PR TITLE
APIS-6703: Remove enumeratum library

### PR DESCRIPTION
**Pull request by johnsgp**

### CODE REVIEW

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/applications/TermsOfUseStatus.scala [hunk 1]
```diff
@@ -16,14 +16,12 @@
...
-object TermsOfUseStatus extends PlayEnum[TermsOfUseStatus] {
-  val values = findValues
+sealed trait TermsOfUseStatus
 
+object TermsOfUseStatus {
   final case object NOT_APPLICABLE     extends TermsOfUseStatus
   final case object AGREEMENT_REQUIRED extends TermsOfUseStatus
   final case object AGREED             extends TermsOfUseStatus
+
+  val values = List(NOT_APPLICABLE, AGREEMENT_REQUIRED, AGREED)
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/applications/TermsOfUseStatus.scala:26]: Ditto

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/connectors/CombinedApi.scala [hunk 2]
```diff
@@ -16,16 +16,27 @@
...
-sealed trait ApiType extends EnumEntry
+sealed trait ApiType
 
-object ApiType extends Enum[ApiType] with PlayJsonEnum[ApiType] {
-  val values = findValues
+object ApiType {
   case object REST_API extends ApiType
   case object XML_API  extends ApiType
+
+  val values = List(REST_API, XML_API)
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/connectors/CombinedApi.scala:29]: Ditto - and anywhere else

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala [hunk 3]
```diff
@@ -38,29 +36,46 @@ object EmailPreferences {
   def noPreferences: EmailPreferences = EmailPreferences(List.empty, Set.empty)
 }
 
-sealed abstract class EmailTopic(val value: String, val displayName: String, val description: String, val displayOrder: Byte) extends StringEnumEntry
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala:41]: Did value really go away?

Comment (AndySpaven) [reply app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala:41]: Also not surer that 4 defs in companion might be better than 4 fields

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/flows/Flow.scala [hunk 4]
```diff
@@ -16,31 +16,36 @@
...
+  val values: List[FlowType] = List(
+    IP_ALLOW_LIST,
+    EMAIL_PREFERENCES,
+    EMAIL_PREFERENCES_V2,
+    NEW_APPLICATION_EMAIL_PREFERENCES,
+    NEW_APPLICATION_EMAIL_PREFERENCES_V2,
+    GET_PRODUCTION_CREDENTIALS
+  )
+
   def from[A <: Flow: TypeTag]: FlowType = {
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/flows/Flow.scala:49]: Can we loose this typetag stuff - bit to java .isInstanceOf for my liking

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala (hunk 3 above)
Comment (AndySpaven) [reply app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala:41]: Ah - Value is the toString equivalent.

File: app/uk/gov/hmrc/apiplatform/modules/mfa/models/MfaModels.scala [hunk 5]
```diff
@@ -50,6 +46,10 @@ object MfaType extends Enum[MfaType] with PlayJsonEnum[MfaType] {
     override def asText: String = "Text message"
   }
 
+  val values: List[MfaType] = List(AUTHENTICATOR_APP, SMS)
```
[the remark this answers is not in the document]
Comment (johnsgp) [reply app/uk/gov/hmrc/apiplatform/modules/mfa/models/MfaModels.scala:49]: Is it important for it to be an ordered set? `List` is used in the domain libraries which was used as the reference point.

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala (hunk 3 above)
Comment (johnsgp) [reply app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala:41]: We tested this manually with local testing, connecting to the backend. Seemed to work fine.

File: app/uk/gov/hmrc/apiplatform/modules/mfa/models/MfaModels.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (AndySpaven) [reply app/uk/gov/hmrc/apiplatform/modules/mfa/models/MfaModels.scala:49]: Oh no.  That's a blow.  Looks like somewhere we messed up the domain stuff.  It should be set (and listset if it needs to be ordered).  It's not functionally going to cause issues (who's going to add things twice) but it's intent.

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala (hunk 3 above)
Comment (AndySpaven) [reply app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/emailpreferences/emailpreferences.scala:41]: Yeah - ignore the value thing - further down I saw how it was being used hence comment 3 of mine.

**Review by AndySpaven — APPROVED** [review]

**Pull request merged**